src/main/java/teammates/ui/webapi/UpdateAccountRequestAction.java

```diff
@@ -0,0 +1,79 @@
+        String institute = getNonNullRequestParamValue(Const.ParamsNames.INSTRUCTOR_INSTITUTION);
+
+        AccountRequestAttributes accountRequest = logic.getAccountRequest(email, institute);
+        if (accountRequest == null) {
+            throw new EntityNotFoundException("Account request with email: " + email
+                    + " and institute: " + institute + " does not exist.");
+        }
+        if (!accountRequest.getStatus().equals(AccountRequestStatus.SUBMITTED)) {
+            throw new InvalidOperationException("Only account requests with status SUBMITTED can be edited.");
+        }
+
+        AccountRequestUpdateRequest updateRequest = getAndValidateRequestBody(AccountRequestUpdateRequest.class);
+        UpdateOptions newAccountRequest = AccountRequestAttributes.updateOptionsBuilder(email, institute)
+                .withName(updateRequest.getInstructorName())
+                .withInstitute(updateRequest.getInstructorInstitute())
+                .withEmail(updateRequest.getInstructorEmail())
+                .withLastProcessedAt(Instant.now())
+                .build();
+
+        try {
+            AccountRequestAttributes updatedAccountRequest = logic.updateAccountRequest(newAccountRequest);
+            taskQueuer.scheduleAccountRequestForSearchIndexing(updatedAccountRequest.getEmail(),
+                    updatedAccountRequest.getInstitute());
+
+            return new JsonResult(new AccountRequestData(updatedAccountRequest));
+        } catch (InvalidParametersException ipe) {
+            throw new InvalidHttpRequestBodyException(ipe);
+        } catch (EntityDoesNotExistException ednee) {
+            throw new EntityNotFoundException("Account request with email: " + email
+                    + " and institute: " + institute + " does not exist.", ednee);
+        } catch (EntityAlreadyExistsException eaee) {
+            AccountRequestAttributes existingAccountRequest =
+                    logic.getAccountRequest(updateRequest.getInstructorEmail(), updateRequest.getInstructorInstitute());
+            throw new InvalidOperationException(generateExistingAccountRequestErrorMessage(
+                    existingAccountRequest.getStatus()), eaee);
+        }
+    }
+
+    private String generateExistingAccountRequestErrorMessage(AccountRequestStatus status) {
+        String basicMessage = "There’s an existing account request with the email address and institute"
+                + " you want to update to and its status is " + status + ".\n";
+        StringBuilder sb = new StringBuilder(basicMessage);
+        switch (status) {
+        case SUBMITTED:
+            sb.append("You can locate that account request on the requests page and process it instead.");
+            break;
+        case APPROVED:
+        case REGISTERED:
+        case REJECTED:
+            sb.append("You can search for that account request to see more information.");
+            break;
+        }
```

Comment:
These messages are not appropriate, backend should not have any knowledge of UI.

Reply:
For this, I wonder how you can achieve the same behavior without doing this.

Reply:
Ok I'll make the change.